react-renderer/src/Constraints.ts

```diff
@@ -3,173 +3,173 @@ import { canvasSize } from "./Canvas";
     // can this be made more efficient (code-wise) by calling "above" and swapping arguments? - stella
 
 
-  centerLabel: ([t1, arr]: [string, any], [t2, text1]: [string, any], w: number): Tensor => {
-    if (typesAre([t1,t2], ["Arrow", "Text"])) {
-      const mx = arr.startX.contents.add(arr.endX.contents).div(scalar(2.0));
-      const my = arr.startY.contents.add(arr.endY.contents).div(scalar(2.0));
-      // entire equation is (mx - lx) ^ 2 + (my + 1.1 * text.h - ly) ^ 2 from Functions.hs - split it into two halves below for readability
-      const lh = mx.sub(text1.x.contents).square();
-      const rh = my.add(text1.h.contents.mul(scalar(1.1))).sub(text1.y.contents).square();
-      return lh.add(rh).mul(w);
-    } else throw new Error(`${[t1, t2]} not supported for centerLabel`)
-  },
-
-  // Generic repel function for two GPIs with centers
-  repel: ([t1, s1]: [string, any], [t2, s2]: [string, any]) => {
-    // HACK: `repel` typically needs to have a weight multiplied since its magnitude is small
-    // TODO: find this out programmatically
-    const repelWeight = 10e6;
-    // 1 / (d^2(cx, cy) + eps)
-    return distsq(center(s1), center(s2)).add(epsd).reciprocal().mul(repelWeight);
-  },
-
-  centerArrow: ([t1, arr]: [string, any], [t2, text1]: [string, any], [t3, text2]: [string, any]): Tensor => {
-    const spacing = scalar(1.1); // arbitrary
-
-    if (typesAre([t1, t2, t3], ["Arrow", "Text", "Text"])) {
-      // HACK: Arbitrarily pick the height of the text
-      // [spacing * getNum text1 "h", negate $ 2 * spacing * getNum text2 "h"]
-      return centerArrow2(arr, center(text1), center(text2),
-        [spacing.mul(text1.h.contents),
-        text2.h.contents.mul(spacing).mul(scalar(1.0)).neg()]);
-    } else throw new Error(`${[t1, t2, t3]} not supported for centerArrow`);
-  },
+    centerLabel: ([t1, arr]: [string, any], [t2, text1]: [string, any], w: number): Tensor => {
+        if (typesAre([t1, t2], ["Arrow", "Text"])) {
+            const mx = arr.startX.contents.add(arr.endX.contents).div(scalar(2.0));
+            const my = arr.startY.contents.add(arr.endY.contents).div(scalar(2.0));
+            // entire equation is (mx - lx) ^ 2 + (my + 1.1 * text.h - ly) ^ 2 from Functions.hs - split it into two halves below for readability
+            const lh = mx.sub(text1.x.contents).square();
+            const rh = my.add(text1.h.contents.mul(scalar(1.1))).sub(text1.y.contents).square();
+            return lh.add(rh).mul(w);
+        } else throw new Error(`${[t1, t2]} not supported for centerLabel`)
+    },
+
+    // Generic repel function for two GPIs with centers
+    repel: ([t1, s1]: [string, any], [t2, s2]: [string, any]) => {
+        // HACK: `repel` typically needs to have a weight multiplied since its magnitude is small
+        // TODO: find this out programmatically
+        const repelWeight = 10e6;
+        // 1 / (d^2(cx, cy) + eps)
+        console.log("repel", t1, s1, "shape2", t2, s2);
```

Comment:
Remember to comment this out. Loosely related: maybe we should start using loggers.

Reply:
Fixed